docusaurus/docs/React/custom-code-examples/video-integration-100ms.mdx

```diff
@@ -0,0 +1,881 @@
+```
+
+Great, with that we are prepared to implement the `createCall` function. We will do four things:
+
+1. Set the state to `ConnectionState.Connecting`
+2. Create a call from the `channel` object
+3. Join the call using the `hmsActions`
+4. Update the channel data (which will trigger an update of our UI due to the `channel.updated` event being called)
+
+Inside of the `store` object replace the empty `createCall` implementation with this:
+
+```tsx
+createCall: async () => {
+  // 1.
+  setConnectionState(ConnectionState.Connecting);
+
+  // 2.
+  const response = await channel.createCall({
+    id: `call-${channel.cid}`,
+    type: 'video',
+  });
+
+  // 3.
+  await hmsActions.join({
+    authToken: `${response.token}`,
+    userName: client.user?.name ?? 'Unkown',
+  });
+
+  // 4.
+  await channel.updatePartial({
+    set: {
+      data: {
+        callActive: true,
+        callId: response.call.id,
+      },
+    },
+  });
+},
+```
+
+The `joinCall` function is pretty similar to the `createCall` function only that we’re now taking the `callId` we save in the channel `data` object and using it to call the `getCallToken` function on the `client` object. Then we join the call as before. Here is the code to replace the current (empty) `joinCall` function with:
+
+```tsx
+joinCall: async () => {
+  setConnectionState(ConnectionState.Connecting);
+
+  const response = await client.getCallToken(channel.data?.data?.callId);
+
+  await hmsActions.join({
+    authToken: `${response.token}`,
+    userName: client.user?.name ?? 'Unkown',
+  });
+
+  setConnectionState(ConnectionState.InCall);
+},
+```
+
+When leaving a call the logic is quite straightforward. We’ll set the state to `ConnectionState.Disconnecting`, then calling the `leave` function of the `hmsActions` SDK, and then updating the state to `ConnectionState.CallAvailable`. Here is the code:
+
+```tsx
+leaveCall: async () => {
+  setConnectionState(ConnectionState.Disconnecting);
+
+  await hmsActions.leave();
+
+  setConnectionState(ConnectionState.CallAvailable);
+},
+```
+
+The last function is the `endCall` which is similar to the `leaveCall` but in addition, we also need to update the channel `data` object. With that, we don’t need to manually update the connection state because this will be again handled by our subscription to the `channel.updated` event.
+
+Replace the current `endCall` function with this one:
+
+```tsx
+endCall: async () => {
+  setConnectionState(ConnectionState.Disconnecting);
+
+  await hmsActions.leave();
+
+  await channel.updatePartial({
+    set: {
+      data: {
+        callActive: false,
+      },
+    },
+  });
+};
+```
+
+With that we have all the logic available, there is just one more thing we’ll add. In order for us to not be stuck in a call when the window is closed, we’ll also leave the call when the `window.onunload` event is happening:
+
+```tsx
+useEffect(() => {
+  window.onunload = () => {
+    hmsActions.leave();
+  };
+}, []);
```

Comment:
```suggestion
useEffect(() => () => {  
    hmsActions.leave();
}, []);
```
We can utilise React's cleaning function which gets invoked whenever either dependency changes or the component itself unmounts.

Reply:
Now also thinking about `hmsActions` - is that object stable? So that when the cleaning function gets invoked it doesn't call `leave` on a "stale" `hmsActions` reference.

Reply:
Object should be stable. That is a weird syntax returning from useEffect with only a function but makes sense. :)